### PR TITLE
[Backport 2.x] Add a new configuration setting `synonym_analyzer` for `synonym_graph` and synonym

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Backport of https://github.com/opensearch-project/OpenSearch/pull/16488.

### Related Issues
Part of https://github.com/opensearch-project/OpenSearch/issues/16263 and https://github.com/opensearch-project/OpenSearch/issues/16530.

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
             </list>
         </option>
     </component>
-  <component name="VcsDirectoryMappings">
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
-  </component>
+    <component name="VcsDirectoryMappings">
+        <mapping directory="$PROJECT_DIR$" vcs="Git" />
+    </component>
```
Comment (owaiskazi19) [inline .idea/vcs.xml:19]: We should not commit this

**Review by owaiskazi19 — CHANGES_REQUESTED** [review]

File: .idea/vcs.xml (hunk 1 above)
Comment (prudhvigodithi) [reply .idea/vcs.xml:19]: @owaiskazi19 this is already part of `main` and `2.x` branches, should we completely remove that or keep it as it is?, this change is made by local IDE. I havent intentionally made any change.

Comment (owaiskazi19) [reply .idea/vcs.xml:19]: Yes, the file is there. I meant the formatting change. Let's remove this change from the commit.

Comment (prudhvigodithi) [reply .idea/vcs.xml]: Done, can you please check the PR again?

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**